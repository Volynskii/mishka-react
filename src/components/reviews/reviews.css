.reviews {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 1050px;
    margin: 0;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 67px;
    height: 457px;
    border: 1px solid;
    background-image: url("./images/review.png");
    background-repeat: no-repeat;
    background-position: left;
}
.reviews__wrapper {
    height: 275px;
    width: 876px;
    position: relative;
}
.reviews__headline {
    position: relative;
    bottom: 8px;
    font-size: 53px;
    line-height: 53px;
    font-weight: bold;
    margin: 0;
    padding: 0;
    text-align: right;
}
.reviews__button {
    color:#444444;
    font-size: 17px;
    line-height: 20px;
    font-weight: bold;
    background-color:transparent;
    width: 170px;
    height: 48px;
    margin: 0;
    margin-left: auto;
    padding: 0;
    position: absolute;
    top: 0;
    right: 0;
}
.reviews__comment {
    font-size: 17px;
    line-height: 30px;
    margin: 0;
    margin-top: 35px;
}
.reviews__author {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 45px;
    position: absolute;
    left: 0;
    bottom: 10px;
    list-style-type: none;
    margin: 0;
    padding: 0;
}
.reviews__author__name {
    font-size: 17px;
    font-weight: bold;
}
.reviews__slider {
    width: 137px;
    height: 45px;
    display: flex;
    justify-content: space-between;
    position: absolute;
    right: 0;
    bottom: 0;
    list-style-type: none;
    margin: 0;
    padding: 0;
}
.reviews__slider__left {
width: 50px;
    background-image: url("./images/huge-arrow-left.png");
    background-position: top;
    background-repeat: no-repeat;
}
.reviews__slider__right {
    width: 50px;
    background-image: url("./images/huge-arrow-right.png");
    background-position: top;
    background-repeat: no-repeat;
}

