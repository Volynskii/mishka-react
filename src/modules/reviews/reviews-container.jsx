import React, {useState} from "react";
import { useSelector } from "react-redux";
import Reviews from "./reviews/reviews";
import {FormReviews} from "../../components/form-reviews/form-reviews";

export const ReviewsContainer = () => {

    const commentsData = useSelector((state) => state.comments.comments);
    const indexData = useSelector((state) => state.comments.index);

    const [comments, setComments] = useState(commentsData);
    const [index, setIndex] = useState(indexData);

    const [display,setDisplay] = useState("none");


    const toggleDisplay = () => {
        if (display === "none") {
            setDisplay('flex')
        } else {
            setDisplay("none")
        }

    };
    const nextComment = () => {

        if(index === comments.length - 1) {
             return  setComments(comments)
        }
        setIndex(index + 1);

        };

    const prevComment = () => {
        if(index === 0) {
               return setComments(comments);
        }
        setIndex(index - 1);
    };

    return (
        <>
        <Reviews
        comments={comments}
        index={index}
        nextComment={nextComment}
        prevComment={prevComment}
        toggleDisplay={toggleDisplay}
    />
    <FormReviews
    isOpened={display}
    />
            </>
    )

};
