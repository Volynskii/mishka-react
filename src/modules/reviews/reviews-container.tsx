import React, {useEffect, useState} from "react";
import { useSelector } from "react-redux";
import Reviews from "./reviews/reviews";
import {FormReviews} from "../../components/form-reviews/form-reviews";
import SuccessPopup from "../../components/success-popup/success-popup";
import {rootReducer} from "../../store/reducer";

export const ReviewsContainer = () => {

    const commentsData = useSelector((state: rootReducer) => state.comments.comments);


    const [comments, setComments] = useState(commentsData);
    useEffect(() => {
        setComments(commentsData)
    }, [commentsData]);
    const [index, setIndex] = useState(0);

    const nextComment = () => {
        if(index === comments.length - 1) {
            return;
        }
        setIndex(index + 1);

    };

    const prevComment = () => {
        if(index === 0) {
            return;
        }
        setIndex(index - 1);
    };

    const [display,setDisplay] = useState("none");
    const onOpenPopup = () => {
        setDisplay("flex");
    };

    const onClosePopup = (evt: any) => {
       // const form = document.querySelector('.popup');
       //  form.getElementById('myform').reset();
        setDisplay("none");
    };

    const [successPopupDisplay, setSuccessPopupDisplay] = useState("none");

    const onOpenSuccessPopup = () => {
        setSuccessPopupDisplay("flex");
    };
    const onCloseSuccessPopup = () => {
        setSuccessPopupDisplay("none");
    };
    return (
        <>
            <Reviews
                comments={comments}
                index={index}
                nextComment={nextComment}
                prevComment={prevComment}
                openPopup={onOpenPopup}
            />
            <FormReviews
                isOpened={display}
                closePopup={onClosePopup}
                openSuccessPopup={onOpenSuccessPopup}
            />
            <SuccessPopup
                className={'success-container'}
                textContent={'Ваш отзыв отправлен!'}
                successPopupDisplay={successPopupDisplay}
                closeSuccessPopup={onCloseSuccessPopup}
                title={null}
            />
        </>
    );

};
