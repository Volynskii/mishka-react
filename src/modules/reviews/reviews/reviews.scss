@import "../../../utilities/variables";

/*mobile-mobile-mobile */
@media (min-width: $mobile-width-min) and (max-width: $mobile-width-max) {
    .reviews {
        display: flex;
        justify-content: center;
        align-items: center;
        width: 100%;
        margin: 0;
        margin-left: auto;
        margin-right: auto;
        margin-bottom: 45px;
        padding-top: 20px;
        min-height: 457px;
        background-size: 40% 100%;
        background-repeat: no-repeat;
    }
    .reviews__wrapper {
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: flex-start;
        min-height: 275px;
        width: 82.5%;
        position: relative;
    }
    .reviews__headline {
        order: 0;
        font-size: $headline-font-size-mobile;
        line-height: $headline-font-size-mobile;
        font-weight: bold;
        margin: 0;
        padding: 0;
        text-align: left;
    }

    .reviews__button {
        order:1;
        color: $button-font-color;
        font-size: 17px;
        line-height: 20px;
        font-weight: bold;
        background-color: transparent;
        width: 260px;
        height: 48px;
        margin: 0;
        margin-top: 40px;
        margin-left: auto;
        margin-right: auto;
        padding: 0;
        transition: 0.5s ease-in-out;
    }
    .reviews__button:hover {
        background-color: $button-background-color-hover;
        color: $button-font-color-hover;
    }
    .reviews__button:active {
        opacity: 0.5;
    }
    .reviews__comment {
     display: flex;
        justify-content: center;
        align-items: center;
        width: 100%;
        min-height: 270px;
        max-height: 270px;
        font-size: 15px;
        line-height: 30px;
        margin: 0;
        margin-top: 40px;
        margin-bottom: 40px;
    }
    .reviews__author {
        width: 100%;
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        height: 45px;
        list-style-type: none;
        margin: 0;
        margin-bottom: 40px;
        padding: 0;
    }
    .reviews__author__name {
        font-size: 17px;
        font-weight: bold;
    }
    .reviews__slider {
        width: 100%;
        height: 45px;
        display: flex;
        justify-content: space-between;
        list-style-type: none;
        margin: 0;
        padding: 0;
    }
    .reviews__slider__left {
        border: 1px solid #bdbdbd;
        width: 50%;
        background-color: transparent;
        background-image: url("./images/huge-arrow-left.png");
        background-position: center;
        background-repeat: no-repeat;
        transition: 0.5s ease-in-out;
    }
    .reviews__slider__left:disabled {
        border: 5px solid red;
    }
    .reviews__slider__left:hover {
        background-image: url("./images/huge-arrow-left-hover.png");
    }
    .reviews__slider__left:active {
        background-image: url("./images/huge-arrow-left-active.png");
    }
    .reviews__slider__right {
        background-color: transparent;
        width: 50%;
        background-image: url("./images/huge-arrow-right.png");
        background-position: center;
        background-repeat: no-repeat;
        transition: 0.5s ease-in-out;
        border:0;
        border-top:  1px solid #bdbdbd;
        border-bottom:1px solid #bdbdbd;
        border-right: 1px solid #bdbdbd;
    }
    .reviews__slider__right:hover {
        background-image: url("./images/huge-arrow-right-hover.png");
    }
    .reviews__slider__right:active {
        background-image: url("./images/huge-arrow-right-active.png");
    }
};


/*tablet-tablet-tablet *//*tablet-tablet-tablet */
@media (min-width: $tablet-width-min) and (max-width: $tablet-width-max) {
    .reviews {
        display: flex;
        justify-content: center;
        align-items: center;
        width: 100%;
        margin: 0;
        margin-left: auto;
        margin-right: auto;
        margin-bottom: 70px;
        padding-top: 80px;
        padding-bottom: 80px;
        min-height: 457px;
        //border: 1px solid;
        background-image: url("./images/review.png");
        background-size: 40% 100%;
        background-repeat: no-repeat;
       // background-position: 8% ;
    }
    .reviews__wrapper {
       min-height: 275px;
        width: 87%;
        position: relative;
        //margin-bottom: 80px;
        //margin-top: 80px;
    }
    .reviews__headline {
        position: relative;
        bottom: 8px;
        font-size: $headline-font-size-tablet;
        line-height: $headline-font-size-tablet;
        font-weight: bold;
        margin: 0;
        padding: 0;
        text-align: right;
    }

    .reviews__button {
        color: $button-font-color;
        font-size: 17px;
        line-height: 20px;
        font-weight: bold;
        background-color: transparent;
        width: 170px;
        height: 48px;
        margin: 0;
        margin-left: auto;
        padding: 0;
        position: absolute;
        top: 0;
        right: 0;
        transition: 0.5s ease-in-out;
    }
    .reviews__button:hover {
        background-color: $button-background-color-hover;
        color: $button-font-color-hover;
    }
    .reviews__button:active {
        opacity: 0.5;
    }
    .reviews__comment {
        max-height: 910px;
        font-size: 17px;
        //font-size: 5px;
        line-height: 30px;
        margin: 0;
        margin-top: 40px;
        margin-bottom: 85px;
    }
    .reviews__author {
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        height: 45px;
        position: absolute;
        left: 0;
        bottom: 10px;
        list-style-type: none;
        margin: 0;
        padding: 0;
    }
    .reviews__author__name {
        font-size: 17px;
        font-weight: bold;
    }
    .reviews__slider {
        width: 137px;
        height: 45px;
        display: flex;
        justify-content: space-between;
        position: absolute;
        right: 0;
        bottom: 0;
        list-style-type: none;
        margin: 0;
        padding: 0;
    }
    .reviews__slider__left {
        border: 0;
        width: 50px;
        background-color: transparent;
        background-image: url("./images/huge-arrow-left.png");
        background-position: center;
        background-repeat: no-repeat;
        transition: 0.5s ease-in-out;
    }
    .reviews__slider__left:disabled {
        border: 5px solid red;
    }
    .reviews__slider__left:hover {
        background-image: url("./images/huge-arrow-left-hover.png");
    }
    .reviews__slider__left:active {
        background-image: url("./images/huge-arrow-left-active.png");
    }
    .reviews__slider__right {
        width: 50px;
        background-image: url("./images/huge-arrow-right.png");
        background-position: center;
        background-repeat: no-repeat;
        transition: 0.5s ease-in-out;
        background-color: transparent;
        border: 0;
    }
    .reviews__slider__right:hover {
        background-image: url("./images/huge-arrow-right-hover.png");
    }
    .reviews__slider__right:active {
        background-image: url("./images/huge-arrow-right-active.png");
    }
};


/*desktop-desktop-desktop */
@media (min-width: $desktop-width-min) {
    .reviews {
        display: flex;
        justify-content: center;
       // align-items: center;
        width: 1050px;
        min-height: 457px;
        margin: 0;
        margin-left: auto;
        margin-right: auto;
        margin-bottom: 70px;
        padding-top: 80px;
        padding-bottom: 80px;
        //border: 1px solid;
        background-image: url("./images/review.png");
        background-size: 35%;
        background-repeat: no-repeat;
        background-position: left;
    }
    .reviews__wrapper {
        min-height: 275px;
        width: 876px;
        position: relative;
    }
    .reviews__headline {
        position: relative;
        bottom: 8px;
        font-size: $headline-font-size-desktop;
        line-height: $headline-font-size-desktop;
        font-weight: bold;
        margin: 0;
        padding: 0;
        text-align: right;
    }
    .reviews__button {
        color: #444444;
        font-size: 17px;
        line-height: 20px;
        font-weight: bold;
        background-color: transparent;
        width: 170px;
        height: 48px;
        margin: 0;
        margin-left: auto;
        padding: 0;
        position: absolute;
        top: 0;
        right: 0;
        transition: 0.5s ease-in-out;
    }
    .reviews__button:hover {
        background-color: $button-background-color-hover;
        color: $button-font-color-hover;
    }
    .reviews__button:active {
        opacity: 0.5;
    }
    .reviews__comment {
        font-size: 17px;
        line-height: 30px;
        margin: 0;
        margin-top: 35px;
        margin-bottom: 85px;
    }
    .reviews__author {
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        height: 45px;
        position: absolute;
        left: 0;
        bottom: 10px;
        list-style-type: none;
        margin: 0;
        padding: 0;
    }
    .reviews__author__name {
        font-size: 17px;
        font-weight: bold;
    }
    .reviews__slider {
        width: 137px;
        height: 45px;
        display: flex;
        justify-content: space-between;
        position: absolute;
        right: 0;
        bottom: 0;
        list-style-type: none;
        margin: 0;
        padding: 0;
    }
    .reviews__slider__left {
        width: 50px;
        background-image: url("./images/huge-arrow-left.png");
        background-position: center;
        background-repeat: no-repeat;
        background-color: transparent;
        cursor: $button-cursor;
        transition: 0.5s ease-in-out;
        padding: 0;
        border: 0;
    }
    .reviews__slider__left:hover {
        background-image: url("./images/huge-arrow-left-hover.png");
        border: 0;
    }
    .reviews__slider__left:active {
        background-image: url("./images/huge-arrow-left-active.png");
        border: 0;
    }

    .reviews__slider__right {
        background-color: transparent;
        border: 0;
        width: 50px;
        background-image: url("./images/huge-arrow-right.png");
        background-position: center;
        background-repeat: no-repeat;
        cursor: $button-cursor;
        transition: 0.5s ease-in-out;
        padding: 0;
    }
    .reviews__slider__right:hover {
        background-image: url("./images/huge-arrow-right-hover.png");
        border: 0;
    }
    .reviews__slider__right:active {
        background-image: url("./images/huge-arrow-right-active.png");
        border: 0;
    }
};
