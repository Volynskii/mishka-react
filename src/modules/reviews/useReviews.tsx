import {useSelector} from "react-redux";
import {rootReducer} from "../../store/reducer";
import React, {useEffect} from "react";

export const useReviews = () => {

    const commentsData = useSelector((state: rootReducer) => state.comments.comments);

    const customerReviews = [
        {
            comment: 'Я молодая мама,' +
                ' но всё равно фотографии' +
                ' не набирали и близко такого ' +
                'количества лайков, как у ' +
                'популярных инстамамочек.' +
                ' В отчаянии, я накупила ' +
                'аксессуаров и игрушек в ' +
                'Мишке, и мои фотографии ' +
                'сразу стали более стильными,' +
                ' а также набирают больше лайков!',
            author: 'Анастасия Красильникова',
            email: '@misssssiskras'
        },

        {
            comment: 'Я молодой парень и мои мои' +
                ' фотографии обычно не набирали' +
                ' большого количества лайков, ' +
                'но как только я купил игрушки ' +
                'разных форматов цветов и форматов' +
                ' количество лайков пошло в гору!' +
                ' Я очень доволен! ',
            author: 'Михаил Соколов!',
            email: 'sokolov@gmail.com'
        },
        {
            comment: 'Я очень доволен данными' +
                ' товарами,товары имеют' +
                ' чрезвычайно высокое качество,' +
                ' ' +
                'заказал себе, друзьям, бабушке,дедушке, папе и маме!',

            author: 'Илья Перевалов!',
            email: 'perevalov@gmail.com'
        }

    ];

     const [comments, setComments] = React.useState(customerReviews.concat(commentsData));

    useEffect(() => {
        setComments(comments)
    }, [comments]);

    const [index, setIndex] = React.useState(0);

    const nextComment: () => void = () => {

        if (index === comments.length - 1) {
            return;
        }
        setIndex(index + 1);

    };

    const prevComment: () => void = () => {
        if (index === 0) {
            return;
        }
        setIndex(index - 1);
    };

    const [display, setDisplay] = React.useState<string>("none");

    const onOpenPopup: () => void = () => {
        setDisplay("flex");
    };

    const onClosePopup: () => void = () => {
        setDisplay("none");
    };


    const [successPopupDisplay, setSuccessPopupDisplay] = React.useState<string>("none");

    const onOpenSuccessPopup: () => void = () => {
        setSuccessPopupDisplay("flex");
    };
    const onCloseSuccessPopup: () => void = () => {
        setSuccessPopupDisplay("none");
    };

    return {
        comments,
        index,
        nextComment,
        prevComment,
        onOpenPopup,
        onClosePopup,
        display,
        successPopupDisplay,
        onOpenSuccessPopup,
        onCloseSuccessPopup
    };
};
