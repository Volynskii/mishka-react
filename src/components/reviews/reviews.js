import './reviews.scss'
import {connect} from "react-redux"
import React, {Component} from "react";


 class Reviews extends Component {
    // state = {
    //     index:0,
    // };

render() {

    console.log( 'APP',this.props)
    const {comments,index,buttonLeftIsDisabled,buttonRightIsDisabled} = this.props;
    return (
        <article className="reviews">
            <div className="reviews__wrapper">
                <span className="reviews__headline">Отзывы</span>
                <button className="reviews__button">Написать</button>

                <p className="reviews__comment">Я молодая мама, но всё равно фотографии не набирали
                    и близко такого количества лайков,
                    как у популярных инстамамочек. В отчаянии,
                    я накупила аксессуаров и игрушек в Мишке,
                    и мои фотографии сразу стали более стильными,
                    а также набирают больше лайков!</p>

                <ul className="reviews__author">
                    <li  className="reviews__author__name">{comments[index].author}</li>
                    <li className="reviews__author__email">@misssssiskras</li>
                </ul>
                <section className="reviews__slider">
                    {/*<ButtonNext   commentsLength={comments.length} />*/}
                    {/*<ButtonPrev  commentsLength={comments.length}/>*/}
                    <button disabled={buttonLeftIsDisabled}   onClick={() => this.props.onNext()} className="reviews__slider__left"/>
                    <button  onClick={() => this.props.onSub()}  className="reviews__slider__right"/>
                </section>


            </div>
        </article>
    )
}
};

 function mapStateToProps(state,) {
     return {
         index: state.index,
         buttonLeftIsDisabled:state.buttonLeftIsDisabled,
         buttonRightIsDisabled:state.buttonRightIsDisabled,

     }
 }
 function mapDispatchToProps(dispatch) {
return {
    onNext: () => dispatch({type: 'NEXT'}),
    onSub: () => dispatch({type: 'PREV'})
}
 }
 export default connect(mapStateToProps,mapDispatchToProps) (Reviews);
