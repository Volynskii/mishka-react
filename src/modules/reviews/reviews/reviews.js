import './reviews.scss'
import React from "react";

export const Reviews = ({comments, index, nextComment, prevComment, toggleDisplay}) => {

    return (
        <article className="reviews">
            <div className="reviews__wrapper">
                <span className="reviews__headline">Отзывы</span>
                <button onClick={toggleDisplay} className="reviews__button">Написать</button>

                <p className="reviews__comment">{comments[index].comment}</p>

                <ul className="reviews__author">
                    <li  className="reviews__author__name">{comments[index].author}</li>
                    <li className="reviews__author__email">{comments[index].email}</li>
                </ul>
                <section className="reviews__slider">
                    <button onClick={prevComment} className="reviews__slider__left"/>
                    <button onClick={nextComment}  className="reviews__slider__right"/>
                </section>
            </div>
        </article>
    )
};


 export default Reviews;
