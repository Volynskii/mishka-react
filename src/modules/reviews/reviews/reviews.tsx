import './reviews.scss'
import React, {FC} from "react";

type Props = {
    comments: {
        comment: string;
        author: string;
        email: string;
    }[];
    index: number;
    nextComment: () => void;
    prevComment: () => void;
    openPopup: () => void;
};
export const Reviews: React.FC<Props> = ({
                                             comments,
                                             index,
                                             nextComment,
                                             prevComment,
                                             openPopup
                                         }) => {


    return (
        <article className="reviews">
            <div className="reviews__wrapper">
                <span className="reviews__headline">Отзывы</span>
                <button onClick={openPopup} className="reviews__button">Написать</button>

                <p className="reviews__comment">{comments[index].comment}</p>

                <ul className="reviews__author">
                    <li className="reviews__author__name">{comments[index].author}</li>
                    <li className="reviews__author__email">{comments[index].email}</li>
                </ul>
                <section className="reviews__slider">
                    <button onClick={prevComment} className="reviews__slider__left"/>
                    <button onClick={nextComment} className="reviews__slider__right"/>
                </section>
            </div>
        </article>
    )
};

export default Reviews;
