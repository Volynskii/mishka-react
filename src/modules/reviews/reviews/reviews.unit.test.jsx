import React from "react";
import {mount, shallow} from "enzyme";
import {Provider, useSelector} from "react-redux";
import {Reviews} from "./reviews";
import {useReviews} from "../useReviews";
import configureStore from "redux-mock-store";
import {act} from "react-dom/test-utils";
import {rootReducer} from "../../../store/reducer";


let comments = [
    {
        comment: 'Я молодая мама,' +
            ' но всё равно фотографии' +
            ' не набирали и близко такого ' +
            'количества лайков, как у ' +
            'популярных инстамамочек.' +
            ' В отчаянии, я накупила ' +
            'аксессуаров и игрушек в ' +
            'Мишке, и мои фотографии ' +
            'сразу стали более стильными,' +
            ' а также набирают больше лайков!',
        author: 'Анастасия Красильникова',
        email: '@misssssiskras'
    },
    {
        comment: 'Я молодой парень и мои мои' +
            ' фотографии обычно не набирали' +
            ' большого количества лайков, ' +
            'но как только я купил игрушки ' +
            'разных форматов цветов и форматов' +
            ' количество лайков пошло в гору!' +
            ' Я очень доволен! ',
        author: 'Михаил Соколов!',
        email: 'sokolov@gmail.com'
    },
    {
        comment: 'Я очень доволен данными' +
            ' товарами,товары имеют' +
            ' чрезвычайно высокое качество,' +
            ' ' +
            'заказал себе, друзьям, бабушке,дедушке, папе и маме!',

        author: 'Илья Перевалов!',
        email: 'perevalov@gmail.com'
    }

];





const initialState = {
    comments:comments,
    index:0
};
const mockStore = configureStore();
const store = mockStore(initialState);

let results;
const renderHook = (hook) => {
    function HookWrapper() {
        results = hook();
        return null;
    }
    mount(
        <Provider store={store}>
            <HookWrapper/>
        </Provider>
    );
    return results;
};
renderHook(useReviews)

const initialProps = {
    comments:results.comments,
    index: results.index,
    nextComment: results.nextComment,
    prevComment: results.prevComment
};
const container = shallow(<Reviews comments={comments} index={0}  />);
/* mocking useSelector on our mock store */
jest
    .spyOn(useReviews, "comments")
    .mockImplementation(state => store.getState());
console.log('INDEX',useReviews().index)
describe('usesample', () => {

    // it('works', () => {
    //    const reviewsHook = renderHook(useReviews)
    //     expect(results.index).toEqual(0)
    //     act(() => {
    //         results.nextComment();
    //     });
    //     expect(results.index).toEqual(1)
    // })
    it('works', () => {

        const btn = container.find(".reviews__slider__right");
        act(() => {
             results.nextComment();
        });
        expect(results.index).toEqual(1)

        act(() => {
            results.prevComment();
        });
        expect(results.index).toEqual(0)

         expect(results.commentsData[0].comment).toBe(comments[0].comment)
// btn.simulate('click')
//         expect(container.find('.reviews__comment').text()).toBe(comments[1].comment)


    })


});




// const initialState = {
//     comments:{comment: '123'}
// };
// const mockStore = configureStore();
// const store = mockStore(initialState);
//
// let results;
// const renderHook = (hook) => {
//     function HookWrapper() {
//         results = hook();
//         return null;
//     }
//     shallow(
//         <Provider store={store}>
//             <HookWrapper/>
//         </Provider>
//     );
//     return results;
// };
// renderHook(useReviews)
// const props = {
//    comments: {comment: '123'},
//     index: 0,
//     nextComment: results.nextComment,
//     prevComment: results.prevComment
// };
//  const setUp = (props) => mount(
//      <Provider store={store}>
//          <Reviews comments={{comment:'123'}}/>
//   </Provider>
//  );
//
//  const wrapper = setUp(props);
//
//
// describe('WeekProduct component', ()=> {
//
//     it('should render WeekProduct Component', ()=> {
//         expect(results.index).toEqual(0)
//             expect(wrapper.find('.reviews__comment').html()).not.toBe(null)
//          // expect(wrapper.find('.reviews__comment').text()).toBe(props.comments[0].comment)
//         // expect(wrapper.find('.reviews__author__name').text()).toBe(props.comments[0].author)
//         // expect(wrapper.find('.reviews__author__email').text()).toBe(props.comments[0].email)
//         //
//         // const btn = wrapper.find(".reviews__slider__right")
//         // btn.simulate("click");
//         // expect(0).toEqual(0)
//         //
//         // expect(wrapper.find('.reviews__comment').text()).toBe(props.comments[1].comment)
//         // expect(wrapper.find('.reviews__author__name').text()).toBe(props.comments[1].author)
//         // expect(wrapper.find('.reviews__author__email').text()).toBe(props.comments[1].email)
//     });
//
// });

