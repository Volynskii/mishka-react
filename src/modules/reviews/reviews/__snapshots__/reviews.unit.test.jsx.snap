// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews work should render Reviews Component 1`] = `
<article
  className="reviews"
>
  <div
    className="reviews__wrapper"
  >
    <span
      className="reviews__headline"
    >
      Отзывы
    </span>
    <button
      className="reviews__button"
    >
      Написать
    </button>
    <p
      className="reviews__comment"
    >
      Я молодая мама, но всё равно фотографии не набирали и близко такого количества лайков, как у популярных инстамамочек. В отчаянии, я накупила аксессуаров и игрушек в Мишке, и мои фотографии сразу стали более стильными, а также набирают больше лайков!
    </p>
    <ul
      className="reviews__author"
    >
      <li
        className="reviews__author__name"
      >
        Анастасия Красильникова
      </li>
      <li
        className="reviews__author__email"
      >
        @misssssiskras
      </li>
    </ul>
    <section
      className="reviews__slider"
    >
      <button
        className="reviews__slider__left"
        onClick={[Function]}
      />
      <button
        className="reviews__slider__right"
        onClick={[Function]}
      />
    </section>
  </div>
</article>
`;
